### DISCUSSION TITLE
Make node names translatable

### DISCUSSION THREAD

**Discussion by Kourosh05** [godotengine/godot-proposals#10945] [Editor]
hi everyone! I've been translating the engine for a week or two and recently realized that node names aren't translatable. so i asked in the contributor chat and someone suggested i make a proposal about it. 
I believe having the node names translatable not only unifies the engines UI when in another language, it also makes nodes easier to understand for non-English speakers. whereas right now even when someone uses another language for the editor they need to also know English to some degree in order to work with the engine because beside node names, some translation strings refer to nodes in English as well.

**Comment by theraot** [comment]
The names of the nodes classes are the names of the nodes in the code...

Yes, translating them could make it easier to find them if you don't know them, but I worry it could bring other issues for the users such as making it difficult to find examples online because they are searching the translated name.

What comes to mind is that perhaps Godot could show both translated and untranslated names.

*By the way, I believe the create new node should be able to search the class description, but that is a different proposal.*

**Reply by Kourosh05** [reply]
That's a good idea. We could have the translated version be on the visuals and put the original name on a description note or something

**Reply by Jesusemora** [reply]
Showing the translated and untranslated names would just make it more confusing, for both begginers and advanced users.
As someone who has to deal with translated programming terms, I can tell you they just create barriers between english and non-english speakers and makes learning and finding solutions more dificult, while doubling the number of words to remember.

**Reply by Kourosh05** [reply]
As someone who learned some concepts in another language i can assure you that it doesn't cause confusion. As far as i'm aware what you're refering to is not a translation problem rather a content problem, because when someone uses another language for the engine it is more likely for them to look for tutorials or solutions in that respective language as well. While this does as you said create barriers between English and non-English speakers it would help godot's reach to people who would probably not consider working with it to be attracted to the community(s) that are non-English only, which i believe already exist and if not i would strongly suggest it.

**Comment by timothyqiu** [comment]
Yeah, even though these names are unavoidable when writing scripts, providing a way to show translated names helps non-English speakers, especially primary and middle school students.
